client/views/account/accountProfile.coffee

```diff
@@ -38,7 +46,9 @@ Template.accountProfile.onCreated ->
 
 	@save = ->
 		instance = @
-
+		if !Settings.findOne("Accounts_AllowUsernameChange").value
+			toastr.error t('Username_Change_Disabled')
+			return callback()
```

Comment:
I'll be honest, not 100% sure what the return callback() does just copied from somewhere that looked similar.